src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3612,27 +3547,20 @@ public override SyntaxNode ObjectCreationExpression(SyntaxNode type, IEnumerable
             return SyntaxFactory.ObjectCreationExpression((TypeSyntax)type, CreateArgumentList(arguments), null);
         }
 
-        private ArgumentListSyntax CreateArgumentList(IEnumerable<SyntaxNode> arguments)
+        private static ArgumentListSyntax CreateArgumentList(IEnumerable<SyntaxNode> arguments)
         {
             return SyntaxFactory.ArgumentList(CreateArguments(arguments));
         }
 
-        private SeparatedSyntaxList<ArgumentSyntax> CreateArguments(IEnumerable<SyntaxNode> arguments)
+        private static SeparatedSyntaxList<ArgumentSyntax> CreateArguments(IEnumerable<SyntaxNode> arguments)
         {
-            return SyntaxFactory.SeparatedList(arguments.Select(AsArgument).Cast<ArgumentSyntax>());
+            return SyntaxFactory.SeparatedList(arguments.Select(AsArgument));
```

Comment:
@CyrusNajmabadi There are cases that we pass the method group without wrapping inside lambda. Is this ok?

Also, given https://github.com/dotnet/roslyn/pull/6642 I think we should remove lambdas eventually.